src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Assemblies/RoAssembly.cs

```diff
@@ -41,8 +41,10 @@ protected RoAssembly(MetadataLoadContext loader, int assemblyFileCount)
 
         // Location and codebase
         public abstract override string Location { get; }
+#pragma warning disable 672
```

Comment:
Unsure if this should just be `Obsolete`d as well, or just pragma ignored.

Reply:
You can obsolete this as well and use the same DiagnosticId. If you check out https://github.com/dotnet/runtime/pull/39269, you'll see that we're using a pattern of an `Obsoletions.cs` file to make the descriptions, ids, and the urlformat all reusable in the `src` files.

But you'll want to rebase on top of that PR once it's merged.

Reply:
Once #39269 is merged (very soon), there will be a common `Obsoletions.cs` file now even, where all obsoletion messages/diagnosticid values are maintained, and that file can be linked into all projects that use new obsoletions.

Reply:
Thanks @jeffhandley, I'll rebase and make those changes once that PR is merged

Reply:
Updated to use common Obsoletions class

Reply:
Do I need to obsolete this in a different way to support net462 etc. builds with the Diagnostic ID etc.?

Seems like adding the `[Obsolete]` here is breaking some builds, eg.
https://github.com/dotnet/runtime/pull/39261/checks?check_run_id=886299009

Reply:
It needs `#if NET50_OBSOLETIONS` in this file. Other places should be ok. They are not in components compiled for older runtime versions.